src/osd/scrubber/pg_scrubber.cc

```diff
@@ -466,6 +466,10 @@ void PgScrubber::on_new_interval()
   // That resets both the scrubber and the FSM.
   m_fsm->process_event(IntervalChanged{});
 
+  // If we are a reserved replica - we need to free ourselves; otherwise -
+  // this is a no-op.
+  dec_scrubs_remote();
```

Comment:
I'd rather see this resource managed by the machine so that upper layers don't need specific support for it.  I suggest a ReplicaActive parent state for the ReplicaActiveOp state which persists between ops and has a bool representing whether a reservation is held.

Reply:
OK - I will try that.

Reply:
@athanatos: so, to recap: I'll need to make the following changes:
- pass the request/release to the FSM (I was trying to avoid it, as now no state change will be affected);
- find a trigger to get out of that ReplicaActive state (apart from 'release'). I'll probably use something similar to what you've used in the Crimson FSM

Do you see anything amiss?

Reply:
That looks right -- the trigger to leave the ReplicaActive state would be interval change.